x/profiles/types/chain_link.go

```diff
@@ -0,0 +1,176 @@
+package types
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/gogo/protobuf/proto"
+)
+
+// NewChainConfig is a constructor function for ChainConfig
+func NewChainConfig(name string) ChainConfig {
+	return ChainConfig{
+		Name: name,
+	}
+}
+
+func (chainConfig ChainConfig) Validate() error {
+	if strings.TrimSpace(chainConfig.Name) == "" {
+		return fmt.Errorf("chain name cannot be empty or blank")
+	}
+	return nil
+}
+
+// ___________________________________________________________________________________________________________________
+
+// NewProof is a constructor function for Proof
+func NewProof(pubKey cryptotypes.PubKey, signature string, plainText string) Proof {
+	pubKeyAny, err := codectypes.NewAnyWithValue(pubKey)
+	if err != nil {
+		panic("failed to pack public key to any type")
+	}
+	return Proof{
+		PubKey:    pubKeyAny,
+		Signature: signature,
+		PlainText: plainText,
+	}
+}
+
+func (proof Proof) Validate() error {
+	if proof.PubKey == nil {
+		return fmt.Errorf("public key field cannot be empty")
+	}
+	_, err := hex.DecodeString(proof.Signature)
+	if err != nil {
+		return fmt.Errorf("failed to decode hex string of signature")
+	}
+	if strings.TrimSpace(proof.PlainText) == "" {
+		return fmt.Errorf("plain text cannot be empty or blank")
+	}
+	return nil
+}
+
+func (proof Proof) Verify(unpacker codectypes.AnyUnpacker) error {
+	var pubkey cryptotypes.PubKey
+	err := unpacker.UnpackAny(proof.PubKey, &pubkey)
+	if err != nil {
+		return fmt.Errorf("failed to unpack the pubkey")
+	}
+	sig, _ := hex.DecodeString(proof.Signature)
+	if !pubkey.VerifySignature([]byte(proof.PlainText), sig) {
+		return fmt.Errorf("failed to verify the signature")
+	}
+	return nil
+}
+
+// ___________________________________________________________________________________________________________________
+
+type AddressData interface {
+	proto.Message
+	Validate() error
+	GetAddressString() string
```

Comment:
Can you please add the docs to these methods? I think it would be important for developers to know what they should do when implementing them